## 56% of end-end tests have passed
```
Total executed tests: 16
Failed tests: 4
Expected to fail tests: 3
Passed tests: 9
```
```
IDs of failed tests: 727231,740490,703133,702843 
```
```
IDs of expected to fail tests: 741840,741839,741841 
```
<h3>Failed tests (4)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=tests/testnetwork-tx-refactor&PR_ID=22352&APK_NAME=22352.apk&TR_CASE_IDS=727231,740490,703133,702843">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823786">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><p><blockquote><div>Device 1: Wait for text element `EmojisNumber` to be equal to `1` </div><div>Device 1: Find `EmojisNumber` by `xpath`: `//*[starts-with(@text,'Message AFTER edit 2 (Edited)')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']/../..//*[@content-desc='emoji-reaction-2']/android.widget.TextView[2]` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:388: in test_community_message_edit
    self.errors.verify_no_errors()
base_test_case.py:209: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: Message reaction is not shown for the sender
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwULNLtDJKwKzZAiA">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjPEO9l2fZjC5n2tZ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823787">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:246: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:209: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: zQ3...dWXh5 was not restored as a contact from waku backup!
E    Device 1: zQ3...Vacac was not restored as a contact from waku backup!
E    Device 1: admin_open was not restored from waku-backup!!
E    Device 1: member_open was not restored from waku-backup!!
E    Device 1: admin_closed was not restored from waku-backup!!
E    Device 1: member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESLqA75XlyfrSBr2Ei">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823790">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><p><blockquote><div># STEP: Adding new regular account</div><div>Device 1: Find `Button` by `accessibility id`: `add-account` </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:357: in test_wallet_add_remove_regular_account
    self.wallet_view.add_regular_account(account_name=new_account_name)
../views/wallet_view.py:252: in add_regular_account
    self.add_account_button.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `add-account` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823791">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/wallet/test_wallet_mainnet.py:70: in test_wallet_balance_mainnet
    self.errors.verify_no_errors()
base_test_case.py:209: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 in total
E    Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 on Optimism
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823796">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td>

```
critical/wallet/test_collectibles.py:100: in test_wallet_send_collectible
    self.wallet_view.reopen_app(user_name=self.sender_username)
../views/base_view.py:348: in reopen_app
    app_package = self.driver.current_package
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/extensions/android/common.py:74: in current_package
    return self.assert_extension_exists(ext_name).execute_script(ext_name)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:405: in execute_script
    return self.execute(command, {"script": script, "args": converted_args})["value"]
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: Could not retrieve the currently focused package and activity. Original error: Error executing adbExec. Original error: 'Command '/root/lambda/shared-tools/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys window displays' exited with code 1'; Command output: adb: device 'emulator-5554' not found
E   
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not retrieve the currently focused package and activity. Original error: Error executing adbExec. Original error: 'Command '/root/lambda/shared-tools/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell dumpsys window displays' exited with code 1'; Command output: adb: device 'emulator-5554' not found
E   
E       at getResponseForW3CError (/root/lambda/shared-tools/volume/appium/appium-2.1.3/node_modules/@appium/base-driver/lib/protocol/errors.js:1145:9)
E       at asyncHandler (/root/lambda/shared-tools/volume/appium/appium-2.1.3/node_modules/@appium/base-driver/lib/protocol/protocol.js:491:57) 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823795">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: CollectibleItemElement </div><div>Device 1: Find `BaseElement` by `xpath`: `//*[@content-desc='expanded-collectible']//android.widget.ImageView` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:81: in test_wallet_collectibles_balance
    if self.wallet_view.expanded_collectible_image.is_element_differs_from_template(
../views/base_element.py:259: in is_element_differs_from_template
    difference = ImageChops.difference(self.image, self.template)
../views/base_element.py:237: in image
    return Image.open(BytesIO(base64.b64decode(self.find_element().screenshot_as_base64)))
../views/base_element.py:83: in find_element
    raise exception
../views/base_element.py:76: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at runMicrotasks (<anonymous>)
E       at processTicksAndRejections (node:internal/process/task_queues:96:5)
E       at AndroidUiautomator2Driver.doFindElementOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/find.js:44:7)
E       at doFind (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:48:17)
E       at wrappedCondFn (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:71:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12) 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823797">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `collectibles-tab` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:157: in test_wallet_collectible_send_from_expanded_info_view
    self.wallet_view.collectibles_tab.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:83: in find_element
    raise exception
../views/base_element.py:76: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at runMicrotasks (<anonymous>)
E       at processTicksAndRejections (node:internal/process/task_queues:96:5)
E       at AndroidUiautomator2Driver.doFindElementOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/find.js:44:7)
E       at doFind (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:48:17)
E       at wrappedCondFn (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:71:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12) 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (9)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823785">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMfPRbwfDbbMO0apL">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRYSnA1M5f0Ci6TTF">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823784">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESLqA75XlyfrSBr2Ei">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823798">test_send_eth[Arbitrum-0.0001]</a>, id: 742015 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823789">test_wallet_send_erc20_from_drawer[Optimism-USD Coin-USDC-2-0.01]</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823799">test_wallet_send_erc20_from_drawer[Mainnet-USD Coin-USDC-2-0.01]</a>, id: 742016 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823788">test_send_eth[Status Network-0.0002]</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823793">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823794">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3823792">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/test/appium/tests/base_test_case.py
+++ b/test/appium/tests/base_test_case.py
@@ -167,12 +167,42 @@ class Driver(webdriver.Remote):
 
 class Errors:
     def __init__(self):
-        self.errors = list()
+        self.errors = []
 
-    def append(self, view: BaseView, text: str):
-        error_text = "Device %s: %s" % (view.driver.number, text)
+    def _add_error(self, view: BaseView, text):
+        """
+        Adds an error message to the list and logs it.
+
+        Args:
+            view (BaseView): The view object containing the driver.
+            text (str): The error message to log.
+        """
+        error_text = f"Device {view.driver.number}: {text}"
         self.errors.append(error_text)
         view.driver.log_event("appium", error_text)
+
+    def append(self, view: BaseView, errors: list | str):
+        """
+        Appends errors to the error list and logs them.
+
+        Args:
+            view (BaseView): The view object containing the driver.
+            errors (str | list): A single error message or a list of error messages.
+
+        Raises:
+            TypeError: If the provided errors argument is not a string or a list of strings.
+        """
+        if isinstance(errors, str):
+            self._add_error(view, errors)
+        elif isinstance(errors, list) and all(isinstance(error, str) for error in errors):
+            for error in errors:
+                self._add_error(view, error)
+        else:
+            raise TypeError(f"Invalid type for errors: {type(errors)}. Expected str or list of str.")
+
+
+
+
 
     def verify_no_errors(self):
         if self.errors:

--- a/test/appium/tests/critical/wallet/test_collectibles.py
+++ b/test/appium/tests/critical/wallet/test_collectibles.py
@@ -39,9 +39,14 @@ class TestWalletCollectibles(MultipleSharedDeviceTestCase):
         self.network_name = 'Base'
 
     @marks.testrail_id(741839)
-    @marks.xfail(reason="Collectibles list is not loaded sometimes")
     def test_wallet_collectibles_balance(self):
         self.wallet_view.collectibles_tab.click()
+        for _ in range(4):
+            self.wallet_view.pull_to_refresh()
+            if not self.wallet_view.element_by_text('No collectibles').is_element_displayed():
+                return
+            time.sleep(5)
+
         self.wallet_view.set_network_in_wallet(self.network_name + ', NEW')
         collectibles = {
             "BVL": {"quantity": 2,
@@ -95,28 +100,27 @@ class TestWalletCollectibles(MultipleSharedDeviceTestCase):
         self.errors.verify_no_errors()
 
     @marks.testrail_id(741840)
-    @marks.xfail(reason="Collectibles list is not loaded sometimes")
     def test_wallet_send_collectible(self):
-        self.wallet_view.reopen_app(user_name=self.sender_username)
+        self.wallet_view.get_account_element().click()
         self.wallet_view.send_button.click()
         self.wallet_view.address_text_input.send_keys(self.receiver['wallet_address'])
         self.wallet_view.continue_button.click()
         self.wallet_view.collectibles_tab_on_select_token_view.click()
-        time.sleep(5)
-        self.wallet_view.get_collectible_element('BVL').click()
-        # self.wallet_view.amount_input_increase_button.click()
-        # self.wallet_view.confirm_button.click()
-        for text in [self.account_name, self.sender_short_address]:
-            if not self.wallet_view.from_data_container.get_child_element_by_text(text).is_element_displayed():
-                self.errors.append(self.wallet_view,
-                                   "Text %s is not shown inside 'From' container on the Review Send page" % text)
-        if not self.wallet_view.to_data_container.get_child_element_by_text(
-                self.receiver_short_address).is_element_displayed():
-            self.errors.append(self.wallet_view,
-                               "Text %s is not shown inside 'To' container on the Review Send page" % text)
-        if not self.wallet_view.on_data_container.get_child_element_by_text(self.network_name).is_element_displayed():
-            self.errors.append(self.wallet_view,
-                               "Text %s is not shown inside 'On' container on the Review Send page" % text)
+        self.wallet_view.get_collectible_element('BVL').wait_and_click(20)
+        self.wallet_view.confirm_button.click()
+        data_to_find_on_review = {
+            self.account_name: ("From", self.wallet_view.from_data_container),
+            self.sender_short_address: ("From", self.wallet_view.from_data_container),
+            self.network_name: ("On", self.wallet_view.on_data_container),
+            self.receiver_short_address: ("To", self.wallet_view.to_data_container)
+        }
+
+        for text, (label, container) in data_to_find_on_review.items():
+            if not container.get_child_element_by_text(text).is_element_displayed():
+                self.errors.append(
+                    self.wallet_view,
+                    "Text %s is not shown inside '%s' container on the Review Send page" % (text, label)
+                )
         data_to_check = {
             'Est. time': r'~\d+ sec',
             'Max fees': r"[$]\d+.\d+",
@@ -151,25 +155,25 @@ class TestWalletCollectibles(MultipleSharedDeviceTestCase):
         self.errors.verify_no_errors()
 
     @marks.testrail_id(741841)
-    @marks.xfail(reason="Collectibles list is not loaded sometimes")
     def test_wallet_collectible_send_from_expanded_info_view(self):
-        # self.wallet_view.reopen_app(user_name=self.sender_username)
         self.wallet_view.collectibles_tab.click()
         self.wallet_view.get_collectible_element('Glitch Punks').wait_for_element().click()
         self.wallet_view.send_from_collectible_info_button.click()
         self.wallet_view.address_text_input.send_keys(self.receiver['wallet_address'])
         self.wallet_view.continue_button.click()
-        for text in [self.account_name, self.sender_short_address]:
-            if not self.wallet_view.from_data_container.get_child_element_by_text(text).is_element_displayed():
-                self.errors.append(self.wallet_view,
-                                   "Text %s is not shown in 'From' container on the Review Send page" % text)
-        if not self.wallet_view.to_data_container.get_child_element_by_text(
-                self.receiver_short_address).is_element_displayed():
-            self.errors.append(self.wallet_view,
-                               "Text %s is not shown in 'To' container on the Review Send page" % text)
-        if not self.wallet_view.on_data_container.get_child_element_by_text(self.network_name).is_element_displayed():
-            self.errors.append(self.wallet_view,
-                               "Text %s is not shown inside 'On' container on the Review Send page" % text)
+        data_to_find_on_review = {
+            self.account_name: ("From", self.wallet_view.from_data_container),
+            self.sender_short_address: ("From", self.wallet_view.from_data_container),
+            self.network_name: ("On", self.wallet_view.on_data_container),
+            self.receiver_short_address: ("To", self.wallet_view.to_data_container)
+        }
+
+        for text, (label, container) in data_to_find_on_review.items():
+            if not container.get_child_element_by_text(text).is_element_displayed():
+                self.errors.append(
+                    self.wallet_view,
+                    "Text %s is not shown inside '%s' container on the Review Send page" % (text, label)
+                )
         data_to_check = {
             'Est. time': r'~\d+ sec',
             'Max fees': r"[$]\d+.\d+",

--- a/test/appium/tests/critical/wallet/test_wallet_testnet.py
+++ b/test/appium/tests/critical/wallet/test_wallet_testnet.py
@@ -1,8 +1,4 @@
-import datetime
-import time
-
 import pytest
-from selenium.common import NoSuchElementException
 
 from base_test_case import MultipleSharedDeviceTestCase, create_shared_drivers
 from support.api.network_api import NetworkApi
@@ -21,7 +17,7 @@ class TestWalletMultipleDevice(MultipleSharedDeviceTestCase):
         self.network_api = NetworkApi()
         self.drivers, self.loop = create_shared_drivers(2)
         self.sign_in_1, self.sign_in_2 = SignInView(self.drivers[0]), SignInView(self.drivers[1])
-        self.sender, self.receiver = transaction_senders['ETH_1'], transaction_senders['ETH_2']
+        self.sender, self.receiver = transaction_senders['ETH_1'], transaction_senders['ETH_3']
         self.sender['wallet_address'] = '0x' + self.sender['address']
         self.receiver['wallet_address'] = '0x' + self.receiver['address']
         self.loop.run_until_complete(
@@ -29,175 +25,79 @@ class TestWalletMultipleDevice(MultipleSharedDeviceTestCase):
                              (self.sign_in_2.recover_access, {'passphrase': self.receiver['passphrase']}))))
         self.home_1, self.home_2 = self.sign_in_1.get_home_view(), self.sign_in_2.get_home_view()
         self.sender_username, self.receiver_username = self.home_1.get_username(), self.home_2.get_username()
-        self.wallet_1, self.wallet_2 = self.sign_in_1.get_wallet_view(), self.sign_in_2.get_wallet_view()
-        self.wallet_1.wallet_tab.click()
-        self.wallet_2.wallet_tab.click()
-        self.network = "Arbitrum"
+        self.wallets = (self.wallet_1, self.wallet_2) = self.sign_in_1.get_wallet_view(), self.sign_in_2.get_wallet_view()
+        [wallet.wallet_tab.click() for wallet in self.wallets]
 
-    def _get_balances_before_tx(self):
-        # ToDo: Arbiscan API is down, looking for analogue
-        # sender_balance = self.network_api.get_balance(self.sender['wallet_address'])
-        # receiver_balance = self.network_api.get_balance(self.receiver['wallet_address'])
-        self.wallet_1.just_fyi("Getting ETH amount in the wallet of the sender before transaction")
-        self.wallet_1.get_account_element().click_if_shown()
-        eth_amount_sender = self.wallet_1.get_asset(asset_name='Ether').get_amount()
-        self.wallet_2.just_fyi("Getting ETH amount in the wallet of the receiver before transaction")
-        self.wallet_2.get_account_element().click_if_shown()
-        eth_amount_receiver = self.wallet_2.get_asset(asset_name='Ether').get_amount()
-        # return sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver
-        return eth_amount_sender, eth_amount_receiver
+    @pytest.mark.parametrize(
+        "network, amount",
+        [
+            pytest.param("Arbitrum", 0.0001, marks=pytest.mark.testrail_id(742015)),
+            pytest.param("Status Network", 0.0002, marks=pytest.mark.testrail_id(727229)),
+        ],
+    )
+    def test_send_eth(self, network, amount):
+        asset, asset_ticker = 'Ether', 'ETH'
+        [wallet.just_fyi("Test to send and verify %s ETH on %s" % (amount, network)) for wallet in self.wallets]
+        eth_amount_receiver = self.wallet_2.get_balance()
+        expected_amount_after_tx_receiver = self.wallet_1.round_amount_float(eth_amount_receiver + amount)
 
-    def _check_balances_after_tx(self, amount_to_send, sender_balance, receiver_balance, eth_amount_sender,
-                                 eth_amount_receiver):
-        # ToDo: Arbiscan API is down, looking for analogue
-        # try:
-        #     self.network_api.wait_for_balance_to_be(address=self.sender['wallet_address'],
-        #                                             expected_balance=sender_balance - amount_to_send)
-        # except TimeoutException as e:
-        #     self.errors.append("Sender " + e.msg)
-        # try:
-        #     self.network_api.wait_for_balance_to_be(address=self.receiver['wallet_address'],
-        #                                             expected_balance=receiver_balance + amount_to_send)
-        # except TimeoutException as e:
-        #     self.errors.append("Receiver " + e.msg)
-
-        def wait_for_wallet_balance_to_update(wallet_view, user_name, initial_eth_amount):
-            wallet_view.just_fyi("Getting ETH amount in the wallet of the %s after transaction" % user_name)
-            if user_name == self.sender_username:
-                exp_amount = round(initial_eth_amount - amount_to_send, 4)
-            else:
-                exp_amount = round(initial_eth_amount + amount_to_send, 4)
-
-            # for _ in range(12):  # ToDo: 120 sec wait time, enable when autoupdate feature is ready
-            wallet_view.get_account_element().click_if_shown()
-            new_eth_amount = round(wallet_view.get_asset(asset_name='Ether').get_amount(), 4)
-            if user_name == self.sender_username and new_eth_amount <= exp_amount:
-                return
-            if user_name == self.receiver_username and new_eth_amount >= exp_amount:
-                return
-            self.errors.append(wallet_view,
-                               "Eth amount in the %s's wallet is %s but should be %s" % (
-                                   user_name, new_eth_amount, exp_amount))
-
-        # ToDo: disable relogin when autoupdate feature is ready
-        self.home_1.just_fyi("Relogin for getting an updated balance")
-        self.home_2.just_fyi("Relogin for getting an updated balance")
-        for _ in range(6):  # just waiting 1 minute here to be sure that balances are updated
-            self.wallet_1.wallet_tab.is_element_displayed()
-            self.wallet_2.wallet_tab.is_element_displayed()
-            time.sleep(10)
-        self.loop.run_until_complete(
-            run_in_parallel(((self.home_1.reopen_app, {'user_name': self.sender_username}),
-                             (self.home_2.reopen_app, {'user_name': self.receiver_username}))))
-        self.wallet_1.wallet_tab.wait_and_click()
-        self.wallet_2.wallet_tab.wait_and_click()
-        self.wallet_1.set_network_in_wallet(network_name=self.network)
-        self.wallet_2.set_network_in_wallet(network_name=self.network)
-        self.loop.run_until_complete(
-            run_in_parallel(((wait_for_wallet_balance_to_update, {'wallet_view': self.wallet_1,
-                                                                  'user_name': self.sender_username,
-                                                                  'initial_eth_amount': eth_amount_sender}),
-                             (wait_for_wallet_balance_to_update, {'wallet_view': self.wallet_2,
-                                                                  'user_name': self.receiver_username,
-                                                                  'initial_eth_amount': eth_amount_receiver}))))
-
-    def _check_last_transaction_in_activity(self, wallet_view, device_time, amount_to_send, sender=True):
-        wallet_view.get_account_element().click()
-        wallet_view.activity_tab.click()
-        wallet_view.just_fyi("Checking the transaction in the activity tab")
-        current_time = datetime.datetime.strptime(device_time, "%Y-%m-%dT%H:%M:%S%z")
-        expected_time = "Today %s" % current_time.strftime('%-I:%M %p')
-        possible_times = [expected_time,
-                          "Today %s" % (current_time + datetime.timedelta(minutes=1)).strftime('%-I:%M %p')]
-        sender_address_short = self.sender['wallet_address'].replace(self.sender['wallet_address'][5:-3], '...').lower()
-        receiver_address_short = self.receiver['wallet_address'].replace(self.receiver['wallet_address'][5:-3],
-                                                                         '...').lower()
-        activity_element = wallet_view.get_activity_element()
-        try:
-            if not all((activity_element.header == 'Send' if sender else 'Receive',
-                        activity_element.timestamp in possible_times,
-                        activity_element.amount == '%s ETH' % amount_to_send,
-                        activity_element.from_text == sender_address_short,
-                        activity_element.to_text == receiver_address_short)):
-                self.errors.append(
-                    wallet_view,
-                    "The last transaction is not listed in activity for the %s, expected timestamp is %s" %
-                    ('sender' if sender else 'receiver', expected_time))
-        except NoSuchElementException:
-            self.errors.append(wallet_view,
-                               "Can't find the last transaction for the %s" % ('sender' if sender else 'receiver'))
-        finally:
-            wallet_view.close_account_button.click_until_presence_of_element(wallet_view.show_qr_code_button)
-
-    @marks.testrail_id(727229)
-    def test_wallet_send_eth(self):
-        self.wallet_1.set_network_in_wallet(network_name=self.network)
-        self.wallet_2.set_network_in_wallet(network_name=self.network)
-        # sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
-        eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
-
-        self.wallet_2.close_account_button.click()
-        self.wallet_2.chats_tab.click()
-
-        self.wallet_1.just_fyi("Sending funds from wallet")
-        amount_to_send = 0.0001
+        self.wallet_1.just_fyi("Sending funds from wallet on %s" % network)
         device_time_before_sending = self.wallet_1.driver.device_time
-        self.wallet_1.send_asset(address='arb1:' + self.receiver['wallet_address'],
-                                 asset_name='Ether',
-                                 amount=f"{amount_to_send:.4f}",
-                                 network_name=self.network)
-        # ToDo: Arbiscan API is down, looking for analogue
-        # self.network_api.wait_for_confirmation_of_transaction(address=self.sender['wallet_address'],
-        #                                                       tx_time=device_time_before_sending)
+        self.wallet_1.get_account_element().click()
+        self.wallet_1.send_asset(address=self.receiver['wallet_address'],
+                                 asset_name=asset,
+                                 amount=self.wallet_1.round_amount_float(amount),
+                                 network_name=network)
 
-        device_time_after_sending = self.wallet_1.driver.device_time
+        self.wallet_1.just_fyi("Verify send tx in the list for sender")
+        tx_errors = self.wallet_1.check_last_transaction_in_activity(device_time_before_sending, amount,
+                                                                     to_account=self.receiver['wallet_address'],
+                                                                     asset=asset_ticker,
+                                                                     tx_type='Send',
+                                                                     network=network)
+        self.errors.append(self.wallet_1, tx_errors)
 
-        # self._check_balances_after_tx(amount_to_send, sender_balance, receiver_balance, eth_amount_sender,
-        #                               eth_amount_receiver)
-        self._check_balances_after_tx(amount_to_send, None, None, eth_amount_sender, eth_amount_receiver)
+        self.wallet_2.just_fyi("Check that balance is updated for receiver")
+        self.wallet_2.wait_for_wallet_balance_to_update(expected_amount_after_tx_receiver)
 
-        # ToDo: enable when issues 20807 and 20808 are fixed
-        # self.loop.run_until_complete(
-        #     run_in_parallel(((self._check_last_transaction_in_activity, {'wallet_view': self.wallet_1,
-        #                                                                  'device_time': device_time,
-        #                                                                  'amount_to_send': amount_to_send}),
-        #                      (self._check_last_transaction_in_activity, {'wallet_view': self.wallet_2,
-        #                                                                  'device_time': device_time,
-        #                                                                  'amount_to_send': amount_to_send,
-        #                                                                  'sender': False}))))
         self.errors.verify_no_errors()
 
-    @marks.testrail_id(727230)
-    def test_wallet_send_asset_from_drawer(self):
+    @pytest.mark.parametrize(
+        "network, asset, asset_ticker, decimals, amount",
+        [
+            pytest.param("Mainnet", 'USD Coin', 'USDC', 2, 0.01,  marks=pytest.mark.testrail_id(742016)),
+            pytest.param("Optimism", 'USD Coin', 'USDC', 2, 0.01,  marks=pytest.mark.testrail_id(727230)),
+        ],
+    )
+    def test_wallet_send_erc20_from_drawer(self, network, asset, asset_ticker, decimals, amount):
+        [wallet.just_fyi("Test to send and verify %s %s on %s from drawer" % (amount, asset, network)) for wallet in self.wallets]
+
         self.wallet_1.navigate_back_to_wallet_view()
-        # sender_balance, receiver_balance, eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
-        eth_amount_sender, eth_amount_receiver = self._get_balances_before_tx()
-        self.wallet_2.close_account_button.click_if_shown()
-        self.wallet_2.chats_tab.click()
+        initial_amount_sender, initial_amount_receiver  = self.wallet_1.get_balance(asset), self.wallet_2.get_balance(asset)
+        expected_amount_after_tx_receiver = self.wallet_1.round_amount_float(initial_amount_receiver + amount, decimals)
+        expected_amount_after_tx_sender = self.wallet_1.round_amount_float(initial_amount_sender - amount, decimals)
 
         self.wallet_1.just_fyi("Sending asset from drawer")
-        amount_to_send = 0.0001
         device_time_before_sending = self.wallet_1.driver.device_time
-        self.wallet_1.send_asset_from_drawer(address='arb1:' + self.receiver['wallet_address'],
-                                             asset_name='Ether',
-                                             amount=f"{amount_to_send:.4f}",
-                                             network_name=self.network)
-        # ToDo: Arbiscan API is down, looking for analogue
-        # self.network_api.wait_for_confirmation_of_transaction(address=self.sender['wallet_address'],
-        #                                                       tx_time=device_time_before_sending)
-        device_time_after_sending = self.wallet_1.driver.device_time
+        self.wallet_1.send_asset_from_drawer(address=self.receiver['wallet_address'],
+                                             asset_name=asset,
+                                             amount=self.wallet_1.round_amount_float(amount, decimals),
+                                             network_name=network)
 
-        # self._check_balances_after_tx(amount_to_send, sender_balance, receiver_balance, eth_amount_sender,
-        #                               eth_amount_receiver)
-        self._check_balances_after_tx(amount_to_send, None, None, eth_amount_sender, eth_amount_receiver)
+        self.wallet_1.just_fyi("Verify send tx in the list for sender")
+        tx_errors = self.wallet_1.check_last_transaction_in_activity(device_time_before_sending, amount,
+                                                                     to_account=self.receiver['wallet_address'],
+                                                                     asset=asset_ticker,
+                                                                     tx_type='Send',
+                                                                     network=network)
+        self.errors.append(self.wallet_1, tx_errors)
 
-        # ToDo: enable when issues 20807 and 20808 are fixed
-        # self.loop.run_until_complete(
-        #     run_in_parallel(((self._check_last_transaction_in_activity, {'wallet_view': self.wallet_1,
-        #                                                                  'device_time': device_time,
-        #                                                                  'amount_to_send': amount_to_send}),
-        #                      (self._check_last_transaction_in_activity, {'wallet_view': self.wallet_2,
-        #                                                                  'device_time': device_time,
-        #                                                                  'amount_to_send': amount_to_send,
-        #                                                                  'sender': False}))))
+        self.wallet_2.just_fyi("Check that balance is updated for receiver")
+        self.wallet_2.wait_for_wallet_balance_to_update(expected_amount_after_tx_receiver, asset, decimals)
+
+        self.wallet_1.just_fyi("Check that balance is updated for sender")
+        self.wallet_1.wait_for_wallet_balance_to_update(expected_amount_after_tx_sender, asset, decimals)
+
         self.errors.verify_no_errors()
+
+

--- a/test/appium/views/base_view.py
+++ b/test/appium/views/base_view.py
@@ -396,3 +396,19 @@ class BaseView(object):
         Button(self.driver, accessibility_id='Wi‑Fi').click()
         self.element_by_text('Done').click()
         self.click_system_back_button_until_presence_of_element(self.chats_tab)
+
+    def pull_to_refresh(self):
+        self.driver.info("Pull to refresh page to get updated data")
+        screen_size = self.driver.get_window_size()
+        width = screen_size["width"] // 2
+        start_y = int(screen_size["height"] * 0.3)
+        end_y = int(screen_size["height"] * 0.7)
+
+        self.driver.execute_script("mobile: swipeGesture", {
+            "left": width,
+            "top": start_y,
+            "width": 10,
+            "height": end_y - start_y,
+            "direction": "down",
+            "percent": 0.75
+        })

--- a/test/appium/views/wallet_view.py
+++ b/test/appium/views/wallet_view.py
@@ -8,6 +8,8 @@ from views.base_element import Button, EditBox, Text, BaseElement
 from views.base_view import BaseView
 from views.home_view import HomeView
 from views.sign_in_view import SignInView
+from typing import Literal
+import datetime
 
 
 class AssetElement(Button):
@@ -76,6 +78,11 @@ class ActivityElement(BaseElement):
     def to_text(self):
         return Text(self.driver, prefix=self.locator,
                     xpath="//*[@content-desc='context-tag'][3]/android.widget.TextView").text
+
+    @property
+    def on_network(self):
+        return Text(self.driver, prefix=self.locator,
+                    xpath="//*[@content-desc='context-tag'][4]/android.widget.TextView").text
 
 
 class ConfirmationViewInfoContainer(BaseElement):
@@ -214,7 +221,7 @@ class WalletView(BaseView):
         for _ in range(3):
             if self.slide_button_track.is_element_displayed():
                 break
-            time.sleep(1)
+            time.sleep(5)
             self.confirm_button.click()
         self.slide_and_confirm_with_password()
 
@@ -222,23 +229,23 @@ class WalletView(BaseView):
         for i in amount:
             Button(self.driver, accessibility_id='keyboard-key-%s' % i).click()
 
-    def send_asset(self, address: str, asset_name: str, amount: str, network_name: str):
+    def send_asset(self, address: str, asset_name: str, amount, network_name: str, account='Account 1'):
         self.send_button.click()
         self.address_text_input.send_keys(address)
         self.continue_button.click()
         self.select_asset(asset_name)
         self.select_network(network_name)
-        self.set_amount(amount)
+        self.set_amount(str(amount))
         self.confirm_transaction()
 
-    def send_asset_from_drawer(self, address: str, asset_name: str, amount: str, network_name: str):
+    def send_asset_from_drawer(self, address: str, asset_name: str, amount, network_name: str):
         asset_element = self.get_asset(asset_name)
         asset_element.long_press_without_release()
         self.send_from_drawer_button.double_click()
         self.select_network(network_name)
         self.address_text_input.send_keys(address)
         self.continue_button.click()
-        self.set_amount(amount)
+        self.set_amount(str(amount))
         self.confirm_transaction()
 
     def add_regular_account(self, account_name: str):
@@ -339,3 +346,78 @@ class WalletView(BaseView):
                 return Text(self.driver, xpath="(%s//android.widget.TextView)[2]" % self.locator).text
 
         return RouteElement(self.driver, route_name)
+
+    @staticmethod
+    def round_amount_float(amount, decimals=4):
+        return round(float(amount), decimals)
+
+
+    def wait_for_wallet_balance_to_update(self, expected_amount, asset='Ether', decimals=4):
+        self.just_fyi(f"Checking {asset} balance, expected value: {expected_amount}")
+
+        start_time = time.time()
+
+        while time.time() - start_time < 120:
+            self.pull_to_refresh()
+            new_balance = self.round_amount_float(self.get_asset(asset_name=asset).get_amount(), decimals)
+
+            # Exit early if the balance is updated
+            if new_balance == expected_amount:
+                return
+
+            time.sleep(10)  # Wait before retrying
+
+        # If the balance is not updated within the timeout, log an error and raise an exception
+        error_message = f"{asset} balance is {new_balance} but expected {expected_amount}"
+        raise TimeoutError(error_message)
+
+    def check_last_transaction_in_activity(self, device_time, amount,
+                                            asset='ETH',
+                                            to_account='',
+                                            tx_type: Literal['Send', 'Receive'] = 'Send',
+                                            from_account_name='Account 1',
+                                            network='Status Network'):
+        errors = list()
+
+        self.just_fyi("Checking the last transaction in the activity tab")
+        current_time = datetime.datetime.strptime(device_time, "%Y-%m-%dT%H:%M:%S%z")
+        expected_time = "Today %s" % current_time.strftime('%-I:%M %p')
+        possible_times = [expected_time,
+                          "Today %s" % (current_time + datetime.timedelta(minutes=1)).strftime('%-I:%M %p')]
+        receiver_variants = [to_account.replace(to_account[5:-3], '...').lower(), 'receiver eth2']
+        activity_element = self.get_activity_element()
+        try:
+            checks = {
+                "header": (activity_element.header, tx_type),
+                "timestamp": (activity_element.timestamp, possible_times),
+                "amount": (activity_element.amount, f"{amount} {asset}"),
+                "from_text": (activity_element.from_text, from_account_name),
+                "to_text": (activity_element.to_text, receiver_variants),
+                "on_network": (activity_element.on_network, network),
+            }
+
+            for name, (left, right) in checks.items():
+                if isinstance(right, list):
+                    if left not in right:
+                        errors.append(
+                            f"Verification of last transaction failed."
+                            f"The failed check: {name} (Expected: {right} to contain: {left})")
+                else:
+                    if left != right:
+                        errors.append(
+                            f"Verification of last transaction failed."
+                            f"The failed check: {name} (Expected: {right}, Found: {left})")
+
+
+
+        except NoSuchElementException:
+            errors.append("Can't find the last transaction")
+        finally:
+            self.close_account_button.click_until_presence_of_element(self.show_qr_code_button)
+        return errors
+
+    def get_balance(self, asset='Ether'):
+        self.just_fyi("Getting %s amount of the wallet of the sender before transaction" % asset)
+        return self.get_asset(asset_name=asset).get_amount()
+
+

